cdap-proto/src/main/java/co/cask/cdap/proto/Id.java

```diff
@@ -71,7 +71,7 @@ private static boolean isValidId(String name) {
     return idPattern.matcher(name).matches();
   }
 
-  private static boolean isValidDatasetId(String datasetId) {
+  public static boolean isValidDatasetId(String datasetId) {
```

Comment:
I was going to use this, but instead opted to use the AbstractVerifier's method, because it is the more restrictive check against dataset names.